zulipterminal/ui_tools/views.py

```diff
@@ -365,27 +370,33 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
             self.set_focus('body')
             self.log.set_focus(self.focus_index_before_search)
             self.view.controller.update_screen()
+            self.in_search_mode = False
             return key
         return_value = super().keypress(size, key)
-        _, self.focus_index_before_search = self.log.get_focus()
+        if not self.in_search_mode:
+            _, self.focus_index_before_search = self.log.get_focus()
```

Comment:
We could just set the focus in the above if condition instead? That would eliminate the need for another attribute, i.e., `in_search_mode`.

Reply:
Oh yes it could be done inside SEARCH_STREAMS right? It was previously outside so I thought it had to be updated everytime. Now, I see there's no use for that.

Reply:
I've made this change and updated the PR